certora/specs/TotalAssetsChange.spec

```
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2025 Morpho Association
+
+methods {
+    function multicall(bytes[]) external => HAVOC_ALL DELETE;
+
+    function accrueInterestView() internal returns(uint256, uint256, uint256) => summaryAccrueInterestView();
+}
+
+// Assume that accrueInterest does nothing.
+function summaryAccrueInterestView() returns (uint256, uint256, uint256) {
+    return (currentContract._totalAssets, 0, 0);
+}
+
+definition mulDivUp(uint256 x, uint256 y, uint256 z) returns mathint = (x * y + (z-1)) / z;
+
+// Check how total assets change without interest accrual.
+rule totalAssetsChange(env e, method f) filtered {
+    f -> !f.isView
+} {
+    mathint totalAssetsPre = currentContract._totalAssets;
+
+    uint256 addedAssets;
+    uint256 removedAssets;
```

```suggestion
    uint256 expectedAddedAssets;
    uint256 expectedrRemovedAssets;
```

While reading, I found the line `require(removedAssets == 0, "this operation only adds assets")` for the `deposit` case to be confusing, for instance I was wondering why isn't this being proved. But this is really the logical reflection of the computation of `_totalAssets`, so by the end it would be nice to be able to read something like expected vs. real.